### PR TITLE
Accept command line options and pass them through.

### PR DESCRIPTION
**Pull request by thisfred**
Some command line options (like `--diff` and `--diff-branch` cannot be
specified in the scalafmt configuration file. This makes a change so
they can be passed in from the pom.xml as follows:

```xml
<plugin>
  <groupId>org.antipathy</groupId>
  <artifactId>mvn-scalafmt</artifactId>
  <version>0.1</version>
  <configuration>
    <parameters>--diff-branch some-branch</paramaters>
    <configLocation>${project.basedir}/path/to/scalafmt.conf</configLocation>
  </configuration>
  ...
</plugin>
```

### CODE REVIEW

**Comment by thisfred** [comment]
Heads up: I *think* this will work, but being a maven ignoramus, I haven't actually figured out how to test this by installing my branch rather than the released version.

**Comment by SimonJPegg** [comment]
Looks good dude.  Couple of things before I merge.

Could you update the readme to include the `<parameters></parameters>` element you've added to the plugin? 

Can you update the `FormatterSpec.scala` test to take a second string argument, empty string is fine.

and do you want to add your details to the `<developers>` section of the pom.xml file? (no is an acceptable answer)

I'm using build profiles in this project so to test you can run

`mvn clean test -Prelease`

Thanks for adding this,
Ciaran.

File: src/main/java/org/antipathy/scalafmtmvn/FormatMojo.java [hunk 1]
```diff
@@ -14,14 +14,16 @@
 
     @Parameter(property="format.configLocation",defaultValue="")
     private String configLocation;
+    @Parameter(property="format.parameters",defaultValue="")
```
Comment (SimonJPegg) [inline src/main/java/org/antipathy/scalafmtmvn/FormatMojo.java:17]: can you update the readme to include an example of this config?

File: src/main/scala/org/antipathy/scalafmtmvn/Formatter.scala [hunk 2]
```diff
@@ -12,11 +15,13 @@ object Formatter {
    * Gets calls scalafmt with the config file specified at configLocation
    * @param configLocation the location of a scalafmt.conf file
    */
-  def format(configLocation: String): Unit =
-    Cli.getConfig(Array[String]("--config", configLocation), CliOptions.default) match {
+  def format(configLocation: String, parameters: String): Unit = {
```
Comment (SimonJPegg) [inline src/main/scala/org/antipathy/scalafmtmvn/Formatter.scala:18]: Can you update the test class with this second param?

**Comment by thisfred** [comment]
Thanks for the feedback! Will address all of these tomorrow. I was able to build and test this today, and see it pick up the parameters specified, so I feel a little bit better about this actually working. ;)

Also: I tried updating the scalafmt version to 0.6.8, the latest released version, because that's what we use at work, and that didn't seem to cause any issues. I can make a separate PR for that if you want.

**Comment by SimonJPegg** [comment]
No include it as part of this.  Could you also bump the version number to 0.2?

**Comment by thisfred** [comment]
Updated with your suggestions, thanks again for the speedy and helpful review!

**Pull request merged**